modules/standard/IO.chpl

```diff
@@ -8936,25 +8956,14 @@ proc fileWriter.writeBinary(const ref data: [?d] ?t, param endian:ioendian = ioe
       err = "writeBinary() array data must be contiguous";
     } else if data.locale.id != this._home.id {
       err = "writeBinary() array data must be on same locale as 'fileWriter'";
-    } else if endian == ioendian.native {
+    } else if endian == ioendian.native || endian == sysEndianness() {
       if data.size > 0 {
         e = try qio_channel_write_amt(false, this._channel_internal, data[d.low], data.size:c_ssize_t * tSize);
       } // else no-op, writing a 0-element array writes nothing
     } else {
       for b in data {
-        select (endian) {
-          when ioendian.native {
-            compilerError("unreachable");
-          }
-          when ioendian.big {
-            e = try _write_binary_internal(this._channel_internal, _iokind.big, b);
-          }
-          when ioendian.little {
-            e = try _write_binary_internal(this._channel_internal, _iokind.little, b);
-          }
-        }
-
-        if e != 0 then break;
```

Comment:
I think this should stick around.

Reply:
Complete oversight, thanks for catching it.  I'm wondering whether a full paratest would've caught that it was missing as well, and whether I should add a test if not.